spotify.py

```diff
@@ -1,51 +1,63 @@
-from billboard import parse_chart
 import requests
 from os import makedirs
 from os.path import exists, join, basename
+import os
 import json
 from operator import itemgetter
+from dotenv import load_dotenv
+
+load_dotenv()
 
 SEARCH_BASE_URL = 'https://api.spotify.com/v1/search?q=/{query}&type=artist'
 RELATED_BASE_URL = "https://api.spotify.com/v1/artists/{id}/related-artists"
 
+AUTH_URL = 'https://accounts.spotify.com/api/token'
+
 
-def fetch_data(artist, url, data_fname, dir_name):
-    if exists(data_fname):
-        pass
-    else:
-        makedirs(dir_name, exist_ok = True)
-        resp = requests.get(url)
-        with open(data_fname, 'wb') as f:
-            f.write(resp.content)
+def fetch_data(url, headers):
+    resp = requests.get(url, headers=headers)
+    return resp.json()
 
-def parse_data(artist, url):
-    dir_name = artist + "_profile"
-    data_fname = dir_name + "id.json"
+def get_related_artists(artist_id, headers):
+    related_url = RELATED_BASE_URL.format(id=artist_id)
+    data = fetch_data(related_url, headers)
 
-    fetch_data(artist, url, data_fname, dir_name)
-    f = open(data_fname, 'r')
```

Comment:
why was I writing this to a file just to then open the file and read a little bit of it?